index.d.ts

```diff
@@ -0,0 +1,49 @@
+// Type definitions for chpr-worker 3.2.0
+// Project: chpr-worker
+// Definitions by: Chauffeur Privé
+// TypeScript Version: 3.0.1
+
+/// <reference types="node" />
+
+import { Logger } from 'chpr-logger';
+
+declare module Worker {
+    export function createWorkers(handlers: Worker.Handler[], config: Worker.Config, options?: Worker.Options): Worker.Process
+}
+
+declare namespace Worker {
+    export interface Process {
+        listen: () => void;
+        stop: () => void;
+        wait: () => Promise<void>;
+        TASK_COMPLETED: string;
+        TASK_RETRIED: string;
+        TASK_FAILED: string;
+        WORKER_CLOSED: string;
+    }
+
+    export interface Handler {
+        handle: (message: string) => void;
```

Comment:
Is the message always a string ?

Reply:
Good catch! message is an object from the bus 🤦‍♀️. @vschoener as it can be anything, can I use the any type here? (same for the validate below)

Reply:
Not any but instead you can define its contract using a generics:
https://www.typescriptlang.org/docs/handbook/generics.html

You can say:
```
export interface Handler<T = any> {
handle: (message: T) = void;
...
}
// And in the usage Code
const handler: Handler<{a: string, b: number}> = {
  handle: (p) => {p.},
};
```

See 

![image](https://user-images.githubusercontent.com/1213200/48419328-ef6eff00-e757-11e8-92ab-24f4216ed7f4.png)

Reply:
Ok thanks a lot for you help! I did as you said 🙏